index.html

```diff
@@ -406,25 +468,195 @@ <h3 id="installation-sec">
+          set when the event is <a data-lt=
+          "construct a BeforeInstallPromptEvent">constructed</a>:
+        </p>
+        <dl>
+          <dt>
+            <dfn>[[\didPrompt]]</dfn>
+          </dt>
+          <dd>
+            A boolean, initially <code>false</code>. Represents whether the
+            event has been used to <a>present an install prompt</a> (either by
+            the developer or by the user agent).
+          </dd>
+          <dt>
+            <dfn>[[\userChoice]]</dfn>
+          </dt>
+          <dd>
+            A promise, which resolves to one of enum values of
+            <a>AppBannerPromptOutcome</a>, which represent the end-user's
+            choice from the UA attempting to <a>present an install prompt</a>.
+          </dd>
+        </dl>
+        <p>
+          To construct an instance of the <a>BeforeInstallPromptEvent</a>
+          interface, run the <dfn data-lt=
+          "construct a BeforeInstallPromptEvent">steps to construct a
+          <code>BeforeInstallPromptEvent</code> event</dfn>:
+        </p>
+        <ol>
+          <li>Let <a>[[\didPrompt]]</a> be <code>false</code>.
+          </li>
+          <li>Let <a>[[\userChoice]]</a> be a newly created promise.
+          </li>
+          <li>If <code>eventInit.userChoice</code> is valid, resolve
+          <a>[[\userChoice]]</a> with <code>eventInit.userChoice</code>.
+          </li>
+        </ol>
+        <section>
+          <h4>
+            <code>prompt()</code> method
+          </h4>
+          <p>
+            The <dfn>prompt</dfn> method, when called, runs the following
+            steps:
+          </p>
+          <ol>
+            <li>If the event's <code>isTrusted</code> attribute is
+            <code>false</code> throw a <a>NotAllowedError</a>, optionally with
+            the message informing the developer that untrusted events can't
+            call <code>prompt()</code>.
+            </li>
+            <li>If the <a>canceled flag</a> is unset, throw an
+            <a>InvalidStateError</a>, optionally with the message informing the
+            developer that <code>.prompt()</code> needs to be called after
+            <code>.preventDefault()</code>.
+            </li>
+            <li>If <a>[[\didPrompt]]</a> is <code>true</code>, throw an
+            <a>InvalidStateError</a>, optionally with the message informing the
+            developer that this event has expired.
+            </li>
+            <li>
+              <a>Request to present an install prompt</a> with this event.
+            </li>
+          </ol>
+          <p>
+            To <dfn>request to present an install prompt</dfn> with
+            <a>BeforeInstallPromptEvent</a> <var>event</var>:
+          </p>
+          <ol>
+            <li>Set <a>[[\didPrompt]]</a> of <var>event</var> to
+            <code>true</code>.
+            </li>
+            <li>
+              <a>In parallel</a>:
```

Comment:
How can this be done in parallel? Step 2.2 is dependent on the results of Step 2.1.